spkrepo/templates/frontend/package.html

```diff
@@ -14,19 +13,38 @@ <h1 class="media-heading">{{ package.versions[-1].displaynames['enu'].displaynam
         </small>
       </h1>
       <p>{{ package.versions[-1].descriptions['enu'].description }}</p>
+      <div class="package-screenshots">
+        {% for screenshot in package.screenshots %}
+        <img class="package-screenshot" src="{{ url_for('nas.data', path=screenshot.path) }}"/>&nbsp;
+        {% endfor %}
+      </div>
+      {% for version in package.versions|reverse %}
       <dl>
+        <dt>Version {{ version.version_string }}</dt>
+        <dd>{{ version.changelog | safe }}</dd>
+        <dt>Date</dt>
+        <dd>{{ version.insert_date }}</dd>
         <dt>Architectures</dt>
         <dd>
-          {% for build in package.versions[-1].builds %}
+          {% for (version, builds) in version.builds_per_dsm.items() %}
+          <!-- Group firmware by DSM or SRM if 1.x -->
+          {% if version == '1' %}
+          SRM {{ version }}.x:
+          {% else %}
+          DSM {{ version }}.x:
+          {% endif %}
+          {% for build in builds %}
           {% for arch in build.architectures %}
-          <span class="label label-default">{{ arch.code }}</span>
+          <a href="https://packages.synocommunity.com/{{ build.path }}"><span class="label label-default">{{ build.firmware.version }} {{ arch.code }}</span></a>
```

Comment:
spkrepo could be used outside SynoCommunity, please replace the URL with some variable or function call. I believe there are some examples on the project elsewhere.

Reply:
Of course, I missed it. I have checked it is `nas.data` for both packages and screenshots.